src/openrct2/paint/Paint.h

```diff
@@ -82,12 +72,28 @@ struct paint_string_struct
     uint8_t* y_offsets;
 };
 
-union paint_entry
+struct paint_entry
 {
+private:
     paint_struct basic;
-    attached_paint_struct attached;
-    paint_string_struct string;
+
+public:
+    paint_struct* AsBasic()
+    {
+        return reinterpret_cast<paint_struct*>(this);
```

Comment:
I'm not a fan as I mentioned on Discord before. you can however make this actually safe by being explicit about it. Consider using ```    std::array<uint8_t, std::max(
        {
            sizeof(paint_struct), 
            sizeof(attached_paint_struct), 
            sizeof(paint_string_struct)
        })> data;``` as the data.

As for the function then.
```suggestion
        auto* res = reinterpret_cast<paint_struct*>(buffer.data());
        ::new ((void*)res) paint_struct();
        return res;
```
This would ensure you have properly initialized members, this approach would be the closest thing to variant without the need of index.

Reply:
ok done. I am surprised this works, or doesn't break bench sprite sort though as these things might use `AsBasic` just to cast and have no intention on re-initialising it.

Reply:
Well glad it works, in the future we should look into splitting this into different pools maybe.